core/server/common/src/main/java/alluxio/ProcessUtils.java

```diff
@@ -67,6 +67,20 @@ public static void fatalError(Logger logger, String format, Object... args) {
    * @param args args for the format string
    */
   public static void fatalError(Logger logger, Throwable t, String format, Object... args) {
+    fatalErrorWithCheck(false, logger, t, format, args);
+  }
+
+  /**
+   * Logs a fatal error and then exits the system if not tolerant master metadata corruption.
+   *
+   * @param tolerantCorruption whether or not to tolerant master metadata corruption
+   * @param logger the logger to log to
+   * @param t the throwable causing the fatal error
+   * @param format the error message format string
+   * @param args args for the format string
+   */
+  public static void fatalErrorWithCheck(boolean tolerantCorruption,
```

Comment:
`ProcessUtils` contains util methods related to running processes, it shouldn't know anything about masters, metadata, or corruption

Reply:
I changed the parameter to `boolean shouldTolerate` means `whether we should tolerate this error and do not exit the system`. What's your opinion on that?

Reply:
I think adding this sort of boolean parameter makes the method API too complex/confusing. If we don't want to trigger a fatal error, we should not call this method in the first place